src/data/services/LmsApiService.js

```diff
@@ -384,6 +385,38 @@ class LmsApiService {
     const url = `${LmsApiService.baseUrl}/enterprise/api/v1/analytics-summary/${enterpriseUUID}`;
     return LmsApiService.apiClient().post(url, formData);
   }
+
+  static updateUserActiveEnterprise = (enterpriseId) => {
+    const url = `${configuration.LMS_BASE_URL}/enterprise/select/active/`;
+    const formData = new FormData();
+    formData.append('enterprise', enterpriseId);
+
+    return LmsApiService.apiClient().post(
+      url,
+      formData,
+    );
+  };
+
+  static fetchEnterpriseLearnerData(options) {
+    const enterpriseLearnerUrl = `${configuration.LMS_BASE_URL}/enterprise/api/v1/enterprise-learner/`;
+    const queryParams = new URLSearchParams({
+      ...options,
+      page: 1,
+    });
+    const url = `${enterpriseLearnerUrl}?${queryParams.toString()}`;
+    return LmsApiService.apiClient().get(url);
+  }
+
+  static async getActiveLinkedEnterprise(username) {
+    const response = await this.fetchEnterpriseLearnerData({ username });
+    const transformedResponse = camelCaseObject(response.data);
+    const enterprisesForLearner = transformedResponse.results;
+    const activeLinkedEnterprise = enterprisesForLearner.find(enterprise => enterprise.active);
+    if (!activeLinkedEnterprise) {
+      logError(`${username} does not have any active linked enterprise customers`);
+    }
+    return activeLinkedEnterprise.enterpriseCustomer;
```

Comment:
Ohh, while edge casey, I believe this would throw a JS error if there is no active linked enterprise. If the conditional on line 415 is true, we can't access the `.enterpriseCustomer` property on `undefined` on this line.

Given this, any caller of `getActiveLinkedEnterprise` should likely handle when `undefined` is returned, e.g.:

```jsx
const activeLinkedEnterprise = await LmsApiService.getActiveLinkedEnterprise(username);
if (!activeLinkedEnterprise) { return; }
if (activeLinkedEnterprise.uuid !== enterpriseId) { ... }
```